caikit/core/data_model/streams/data_stream.py

```diff
@@ -567,6 +597,62 @@ def _from_jsonl_collection_generator(cls, dirname):
         for data_item in DataStream.chain(data_stream_list).flatten():
             yield data_item
 
+    @classmethod
+    def from_multipart_file(cls, filename: str) -> "DataStream[JsonDictValue]":
+        """Loads up a DataStream from a multipart file
+
+            The data items returned in the data stream are determined by the
+            content type for each part in the multipart file by calling
+            the correct DataStream.from_*
+
+        Args:
+            filename (str): Name of file
+
+        Returns:
+            DataStream: Resulting datastream from file
+        """
+        error.value_check(
+            "<COR04987251E>", is_multipart_file(filename), "file is not multipart"
+        )
+        stream_list = []
+        for part in stream_multipart_file(filename):
+            content_type = part.content_type
+            if "json" in content_type:
+                stream_list.append(
+                    cls(cls._from_json_array_buffer_generator, part.fp, part.filename)
+                )
+            elif "csv" in content_type:
+
+                class UtfEncodeIOWrapper(io.IOBase):
```

Comment:
I think we should pull this out. Having it here will redeclare the class on every step of the iteration which is pretty wasteful